app.py

```diff
@@ -1,261 +1,292 @@
+state_code = st.selectbox("State Code", options)
+head_employment_income = st.number_input(
+    "Head Employment Income", min_value=0, step=10000, value=0
+)
+spouse_employment_income = st.number_input(
+    "Spouse Employment Income", min_value=0, step=10000, value=0
+)
+num_children = st.number_input("Number of Children", 0)
+children_ages = {
+    num: st.number_input(f"Child {num} Age", 0) for num in range(1, num_children + 1)
+}
+
+# Submit button
+submit = st.button("Calculate")
+
+
+def create_situation(state_code, head_income, spouse_income=None, children_ages=None):
+    """
+    Create a situation dictionary for the simulation.
+    """
+    if children_ages is None:
+        children_ages = {}
 
-def get_programs(state_code, head_employment_income, spouse_employment_income=None, children_ages = {}):
-    # Start by adding the single head.
     situation = {
         "people": {
             "you": {
                 "age": {YEAR: DEFAULT_AGE},
-                "employment_income": {YEAR: head_employment_income},
+                "employment_income": {YEAR: head_income},
             }
         }
     }
     members = ["you"]
-    if spouse_employment_income is not None:
+    if spouse_income is not None:
         situation["people"]["your partner"] = {
             "age": {YEAR: DEFAULT_AGE},
-            "employment_income": {YEAR: spouse_employment_income},
+            "employment_income": {YEAR: spouse_income},
         }
-        # Add your partner to members list.
         members.append("your partner")
     for key, value in children_ages.items():
         situation["people"][f"child {key}"] = {
             "age": {YEAR: value},
-            "employment_income": {YEAR: 0}
+            "employment_income": {YEAR: 0},
         }
-        # Add child to members list.
         members.append(f"child {key}")
-    # Create all parent entities.
     situation["families"] = {"your family": {"members": members}}
     situation["marital_units"] = {"your marital unit": {"members": members}}
     situation["tax_units"] = {"your tax unit": {"members": members}}
     situation["spm_units"] = {"your spm_unit": {"members": members}}
     situation["households"] = {
         "your household": {"members": members, "state_name": {YEAR: state_code}}
     }
+    return situation
+
+
+def get_programs(
+    state_code,
+    head_employment_income,
+    spouse_employment_income=None,
+    children_ages=None,
+):
+    """
+    Retrieve program calculations for the given situation.
+    """
+    situation = create_situation(
+        state_code, head_employment_income, spouse_employment_income, children_ages
+    )
     simulation = Simulation(situation=situation)
 
-    #benefits breakdown
     benefits_categories = HouseholdBenefits.adds
 
+    taxes_before_refundable_credits = HouseholdTaxBeforeRefundableCredits.adds
+
+    package = "policyengine_us"
+    resource_path_federal = "parameters/gov/irs/credits/refundable.yaml"
```

Comment:
the way we do this in the web app is to navigate the `adds` features of the variables: https://github.com/PolicyEngine/policyengine-app/blob/7e3b1b6117de799f7ef19d199138ea96af93e971/src/pages/household/output/NetIncomeBreakdown.jsx#L176

@nikhilwoodruff do you have an example of doing this in Python?